test/builder/dummies.go

```diff
@@ -549,23 +550,21 @@ func DummyVirtualMachineWebConsoleRequest(namespace, wcrName, vmName, pubKey str
 
 func DummyVirtualMachineSnapshot(namespace, name, vmName string) *vmopv1.VirtualMachineSnapshot {
 	return &vmopv1.VirtualMachineSnapshot{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "VirtualMachineSnapshot",
-			APIVersion: "vmoperator.vmware.com/v1alpha4",
-		},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      name,
 			Namespace: namespace,
-			Finalizers: []string{
```

Comment:
The finalizers are needed for other test

Reply:
Ok, I added it for the convenience  of testing reconcileDeletion. What you mentioned make sense for reconcileNormal. Then changes to deletion are needed